src/main/scala/com/cognite/spark/datasource/DefaultSource.scala

```diff
@@ -158,5 +167,44 @@ class DefaultSource
       case _ => sys.error("Unknown resource type: " + resourceType)
     }
   }
-  // scalastyle:on cyclomatic.complexity method.length
+
+  override def createRelation(
+      sqlContext: SQLContext,
+      mode: SaveMode,
+      parameters: Map[String, String],
+      data: DataFrame): BaseRelation = {
+    val config = parseRelationConfig(parameters)
+    val resourceType = parameters.getOrElse("type", sys.error("Resource type must be specified"))
+
+    val relation = resourceType match {
+      case "events" =>
+        new EventsRelation(config)(sqlContext) //.EventsInsertion(config)
+      case "timeseries" =>
+        new TimeSeriesRelation(config)(sqlContext)
+      case _ => sys.error(s"Resource type '$resourceType does not support savemode")
+    }
+
+    data.foreachPartition(rows => {
+      implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+      val batches = rows.grouped(Constants.DefaultBatchSize).toVector
+
+      config.saveMode match {
+        case "insert" => {
+          batches.parTraverse(relation.insert).unsafeRunSync()
+        }
+        case "upsert" => {
+          batches.parTraverse(relation.upsert).unsafeRunSync()
+        }
+        case "update" => {
+          batches.parTraverse(relation.update).unsafeRunSync()
+        }
+        case "delete" => {
+          batches.parTraverse(relation.delete).unsafeRunSync()
+        }
+        case _ => sys.error("Unknown savemode: " + config.saveMode)
```

Comment:
We should use enums instead and give an error when the user attempts to create a relation with an unsupported savemode, rather than waiting until they try to write to the relation.